src/audio_playback/kdmapi.rs

```diff
@@ -0,0 +1,91 @@
+#[cfg(target_os = "windows")]
+use crate::settings::WasabiSettings;
+#[cfg(target_os = "windows")]
+use std::io::Write;
+
+use crate::{gui::window::WasabiError, utils};
+
+use super::*;
+use kdmapi_rs::{KDMAPIStream, KDMAPI};
+
+pub struct KdmapiPlayer {
+    stream: KDMAPIStream,
+    use_om_list: bool,
+}
+
+impl KdmapiPlayer {
+    pub fn new() -> Result<Self, WasabiError> {
+        let kdmapi = KDMAPI
+            .as_ref()
+            .map_err(|e| WasabiError::SynthError(format!("Failed to load KDMAPI: {e}")))?;
+
+        let stream = kdmapi
+            .open_stream()
+            .map_err(|e| WasabiError::SynthError(format!("Failed to load KDMAPI: {e}")))?;
+
+        Ok(Self {
+            stream,
+            use_om_list: false,
+        })
+    }
+}
+
+impl MidiAudioPlayer for KdmapiPlayer {
+    fn reset(&mut self) {
+        let reset = utils::create_reset_midi_messages();
+        for ev in reset {
+            self.push_event(ev);
+        }
+
+        self.stream.reset();
+    }
+
+    fn push_event(&mut self, data: u32) {
+        self.stream.send_direct_data(data);
+    }
+
+    fn voice_count(&self) -> Option<u64> {
+        None
+    }
+
+    fn configure(&mut self, settings: &SynthSettings) {
+        self.use_om_list = settings.kdmapi.use_om_sflist;
+    }
+
+    #[allow(unused_variables)]
+    fn set_soundfonts(
+        &mut self,
+        soundfonts: &[WasabiSoundfont],
+        _loading_status: Arc<LoadingStatus>,
+        errors: Arc<GuiMessageSystem>,
+    ) {
+        // Due to Windows using UTF16 formatting, this is currently only available for
```

Comment:
I tried searching for something like that for non-Windows but couldn't find something since everything in Rust is UTF8. What I could do is modify the kdmapi crate so it converts the string to UTF16 on Windows and send it as UTF8 normally on other platforms. This should also avoid all cfg flags in Wasabi. Should I go this route?

Reply:
Yeah should be ok to do that, see if there's a way to make an if condition for the platform to avoid using cfg

Reply:
I removed all such conditions from wasabi code and made everything be handled in the kdmapi crate